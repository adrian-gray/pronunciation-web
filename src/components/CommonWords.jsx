import React from "react";

import {
  Paper,
  Table,
  TableBody,
  TableCell,
  TableRow,
  Typography,
  withStyles
} from "@material-ui/core";

import SplitHilite from "./SplitHilite";

const styles = theme => ({
  cell: {
    paddingLeft: 12,
    paddingRight: 12
  },
  largeText: theme.largeText,
  headspace: theme.headspace
});

const CommonWords = props => {
  const { classes, ipa, words } = props;
  const cells = words.map((el, index) => (
    <TableCell className={classes.cell} key={index}>
      <Typography className={classes.largeText}>
        <SplitHilite str={el} key={el} />
      </Typography>
    </TableCell>
  ));

  let currentRow = 0;
  let cellCount = 0;
  const rows = [];
  do {
    rows[currentRow] = (
      <TableRow key={currentRow}>
        {cells[cellCount++]}
        {cells[cellCount] ? cells[cellCount++] : null}
        {cells[cellCount] ? cells[cellCount++] : null}
      </TableRow>
    );
    currentRow++;
  } while (cellCount < cells.length);

  return (
    <div className={classes.headspace}>
      <Typography variant="title" gutterBottom>
        {"Common words with "}
        <SplitHilite str={ipa} />
      </Typography>
      <Paper>
        <Table>
          <TableBody>{rows}</TableBody>
        </Table>
      </Paper>
    </div>
  );
};

export default withStyles(styles)(CommonWords);
