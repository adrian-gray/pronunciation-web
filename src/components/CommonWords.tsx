import React from "react";

import Table from "react-bootstrap/Table";

import SplitHilite from "./SplitHilite";

export default (props: {
  ipa: string;
  words: string[];
  isUserAuth: boolean;
}) => {
  const { ipa, words } = props;
  const cells = words.map((el: string, index: number) => (
    <td className="cell padded-cell" key={index}>
      <p className="large-text">
        <SplitHilite str={el} key={el} />
      </p>
    </td>
  ));

  let currentRow = 0;
  let cellCount = 0;
  const rows = [];
  do {
    rows[currentRow] = (
      <tr key={currentRow}>
        {cells[cellCount++]}
        {cells[cellCount] ? cells[cellCount++] : null}
        {cells[cellCount] ? cells[cellCount++] : null}
      </tr>
    );
    currentRow++;
  } while (cellCount < cells.length);

  return (
    <div className="headspace">
      <h3>
        {"Common words with "}
        <SplitHilite str={ipa} />
      </h3>
      <div className="sub-canvas">
        <Table>
          <tbody>{rows}</tbody>
        </Table>
      </div>
    </div>
  );
};
