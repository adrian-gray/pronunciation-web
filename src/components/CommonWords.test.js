import React from "react";
import { shallow } from "enzyme";
import { render, cleanup } from "@testing-library/react";

import CommonWords from "./CommonWords";

afterEach(cleanup);

test("CommonWords", async () => {
  const props = {
    ipa: "a:",
    words: ["foo", "bar", "baz", "quox", "apple", "banana", "chad", "elle"]
  };

  const { container } = render(<CommonWords {...props} />);

  expect(container).toMatchSnapshot();
});
