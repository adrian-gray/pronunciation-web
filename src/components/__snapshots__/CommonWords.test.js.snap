// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonWords 1`] = `
<div>
  <div
    class="headspace"
  >
    <h3>
      Common words with 
      <span>
        a:
      </span>
       
    </h3>
    <div
      class="sub-canvas"
    >
      <table
        class="table"
      >
        <tbody>
          <tr>
            <td
              class="cell padded-cell"
            >
              <p
                class="large-text"
              >
                <span>
                  foo
                </span>
                 
              </p>
            </td>
            <td
              class="cell padded-cell"
            >
              <p
                class="large-text"
              >
                <span>
                  bar
                </span>
                 
              </p>
            </td>
            <td
              class="cell padded-cell"
            >
              <p
                class="large-text"
              >
                <span>
                  baz
                </span>
                 
              </p>
            </td>
          </tr>
          <tr>
            <td
              class="cell padded-cell"
            >
              <p
                class="large-text"
              >
                <span>
                  quox
                </span>
                 
              </p>
            </td>
            <td
              class="cell padded-cell"
            >
              <p
                class="large-text"
              >
                <span>
                  apple
                </span>
                 
              </p>
            </td>
            <td
              class="cell padded-cell"
            >
              <p
                class="large-text"
              >
                <span>
                  banana
                </span>
                 
              </p>
            </td>
          </tr>
          <tr>
            <td
              class="cell padded-cell"
            >
              <p
                class="large-text"
              >
                <span>
                  chad
                </span>
                 
              </p>
            </td>
            <td
              class="cell padded-cell"
            >
              <p
                class="large-text"
              >
                <span>
                  elle
                </span>
                 
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
